askup/utils/general.py

```diff
@@ -92,6 +116,69 @@ def get_user_place_in_rank_list(user_id):
     return 0
 
 
+def get_user_profile_rank_list(rank_user_id, viewer_user_id):
+    """
+    Aquire and return a list of rows for the user profile rank list.
+    """
+    first_items = get_user_profile_rank_list_elements(viewer_user_id, args=(rank_user_id,))
+    result_row_datas, user_is_present = compose_user_profile_rank_list_row_data(first_items, rank_user_id)
+
+    if user_is_present:
+        return result_row_datas
+
+    result_row_datas += [(0, None, None, None, None)]  # adding an ellipsis row
+    user_item = get_user_profile_rank_list_elements(rank_user_id, 'where ranked.id = %s', (rank_user_id,))
+    user_row_data, _ = compose_user_profile_rank_list_row_data(user_item)
+
+    if not user_row_data:
+        user_row_data = [(-1, rank_user_id, None, None, None)]  # adding an ellipsis row
+
+    result_row_datas += user_row_data
+    return result_row_datas
+
+
+def compose_user_profile_rank_list_row_data(rows, user_id_to_check=None):
+    """
+    Compose user profile rank list row datas from the query result rows.
+    """
+    items = []
+    user_is_present = False
+
+    for place, id, username, first_name, last_name, questions, thumbs_up in rows:
+        if user_id_to_check == id:
+            user_is_present = True
+
+        name = compose_user_full_name(username, first_name, last_name)
+        items.append((place, id, name, questions, thumbs_up))
+
+    return items, user_is_present
+
+
+def compose_user_full_name(username, first_name, last_name):
+    """
+    Return user's full name representation for the views.
+
+    Needed because of unrequired first and last names.
+    """
+    name = (first_name or last_name) and ' '.join((first_name, last_name))
+    return '{} ({})'.format(name, username) if name else username
+
+
+def get_user_profile_rank_list_elements(viewer_user_id, expression='limit 10', args=[]):
```

Comment:
You mustn't define default value for arg as `[]`. It leads to bad situations.

Reply:
done